server/api/src/rs_get_resource_info_for_operation.cpp

```diff
@@ -0,0 +1,85 @@
+#include "irods/get_resource_info_for_operation.h"
+#include "irods/rodsLog.h"
+#include "irods/rsGlobalExtern.hpp"
+#include "irods/rcGlobalExtern.h"
+#include "irods/getRemoteZoneResc.h"
+#include "irods/dataObjCreate.h"
+#include "irods/objMetaOpr.hpp"
+#include "irods/resource.hpp"
+#include "irods/collection.hpp"
+#include "irods/specColl.hpp"
+#include "irods/miscServerFunct.hpp"
+#include "irods/openCollection.h"
+#include "irods/readCollection.h"
+#include "irods/closeCollection.h"
+#include "irods/dataObjOpr.hpp"
+#include "irods/rsGetHostForGet.hpp"
+#include "irods/irods_resource_backport.hpp"
+#include "irods/irods_resource_redirect.hpp"
+#include "irods/irods_at_scope_exit.hpp"
+
+#include <nlohmann/json.hpp>
+
+#include <set>
+#include <cstring>      // for strdup
+#include <optional>
+
+int rs_get_resource_info_for_operation(
+    rsComm_t*     rsComm,
+    dataObjInp_t* dataObjInp,
+    char **       out_info )
+{
+    rodsServerHost_t *rodsServerHost;
+    const int remoteFlag = getAndConnRemoteZone(rsComm, dataObjInp, &rodsServerHost, REMOTE_OPEN);
+    if (remoteFlag < 0) {
+        return remoteFlag;
+    }
+    else if (REMOTE_HOST == remoteFlag) {
+        const int status = rc_get_resource_info_for_operation(rodsServerHost->conn, dataObjInp, out_info);
+        if (status < 0) {
+            return status;
+        }
```

Comment:
The status check is unnecessary.

We can just return the result of the function call.